[bitnami/thanos] Add compactor dataDir setting

gdlx opened the pull request:
### Description of the change

Adds a `dataDir` compactor to customize the data directory.

### Benefits

Allows the setting of a different data directory without overriding the whole args list.

### Possible drawbacks

None

### Applicable issues

Fixes #29835

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

carrodher commented:
Thanks for your contribution! Could you please bump the chart version in the _Chart.yaml_? This is necessary to test the changes and cut a new release.

In the same way, this is just a friendly reminder that signing your commits is important. Your signature certifies that you either authored the patch or have the necessary rights to contribute to the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

gdlx commented:
Hi @carrodher,
I just bumped the chart and fixed the markdown issue.
About the commit signature, it wasn't signed when opening the PR, but I think I fixed it before you asked. Did I do something wrong ?

carrodher commented:
The DCO check is still failing, see https://github.com/bitnami/charts/pull/29856/checks?check_run_id=31396595990 for some tips about how to fix it

gdlx commented:
@carrodher It seems to be all ok now.

dgomezleon commented on bitnami/thanos/Chart.yaml:
```diff
@@ -35,4 +35,4 @@ maintainers:
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/thanos
-version: 15.7.28
+version: 15.7.29
```
```suggestion
version: 15.8.0
```

dgomezleon reviewed (commented):
Thanks for contributing! Please, bump a minor version since it adds a new param.

gdlx commented:
@dgomezleon Is there something else needed for this review ?
It says 1 approving review is required to merge it but I'm not sure if it's all good.

gdlx commented:
Thanks !
